tests/filecheck/dialects/vector/vector_pure_ops.mlir

```diff
@@ -0,0 +1,21 @@
+// RUN: xdsl-opt %s --verify-diagnostics -p cse | filecheck %s
+
+builtin.module {
+  func.func private @vector_cse_test(%0 : memref<4x4xindex>, %1 : vector<1xi1>, %2 : index) {
+    %3 = "vector.load"(%0, %2, %2) : (memref<4x4xindex>, index, index) -> vector<2xindex>
+    "vector.store"(%3, %0, %2, %2) : (vector<2xindex>, memref<4x4xindex>, index, index) -> ()
+    %4 = "vector.broadcast"(%2) : (index) -> vector<1xindex>
+    %5 = "vector.fma"(%3, %3, %3) : (vector<2xindex>, vector<2xindex>, vector<2xindex>) -> vector<2xindex>
+    func.return
+  }
+}
```

Comment:
nit:
```suggestion
%m0, %i0 = "test.op"() : () -> (memref<4x4xindex>, index)
%load = "vector.load"(%m0, %i0, %i0) : (memref<4x4xindex>, index, index) -> vector<2xindex>
"vector.store"(%load, %m0, %i0, %i0) : (vector<2xindex>, memref<4x4xindex>, index, index) -> ()
%broadcast = "vector.broadcast"(%i0) : (index) -> vector<1xindex>
%fma = "vector.fma"(%load, %load, %load) : (vector<2xindex>, vector<2xindex>, vector<2xindex>) -> vector<2xindex>
```
1. we prefer to use test.op to create fake values for tests, and generally minimise the dependencies on other dialects to avoid spurious test breaks
2. adding names to values makes the naming more stable, if you add an op in the middle it doesn't mess up the check lines below
3. the `builtin.module` is implied if not present, makes the test a little easier to read.

Reply:
I think it's ok to merge as-is, bit making these changes would make the test a little more future-proof, even if it tests the same functionality

Reply:
The naming is definitely clearer. Do you want me to rebase this change in to the last commit?

Reply:
Ah I see, need to update the checks anyway. Will do that and then squash the test changes together

Reply:
updated the checks and squashed all the test commits into one